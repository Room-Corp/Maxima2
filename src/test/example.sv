`timescale 1ns / 1ps
`default_nettype none

// a simple module that will increment c_out on the clock edge if a_in < b_in
// otherwise, c_out doesn't change.
module example (
        input wire          clk_in,
        input wire          rst_in,
        input wire [5:0]    a_in,
        input wire [5:0]    b_in,
        output logic [4:0]  c_out);

  always_ff @(posedge clk_in) begin
    if (rst_in) begin
      // reset to 0
      c_out <= 0;
    end else begin
      c_out <= (a_in > b_in) ? c_out : c_out + 1;
    end
  end
endmodule

`default_nettype wire
