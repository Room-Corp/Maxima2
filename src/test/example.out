#! /opt/homebrew/Cellar/icarus-verilog/12.0/bin/vvp
:ivl_version "12.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/system.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/vhdl_sys.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/vhdl_textio.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/v2005_math.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/va_math.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/v2009.vpi";
S_0x138f06cf0 .scope package, "$unit" "$unit" 2 1;
 .timescale 0 0;
S_0x138f04080 .scope module, "example_tb" "example_tb" 3 3;
 .timescale -9 -12;
v0x6000017543f0_0 .var "a_in", 5 0;
v0x600001754480_0 .var "b_in", 5 0;
v0x600001754510_0 .net "c_out", 4 0, v0x600001754240_0;  1 drivers
v0x6000017545a0_0 .var "clk_in", 0 0;
v0x600001754630_0 .var "rst_in", 0 0;
S_0x138f041f0 .scope begin, "$ivl_for_loop0" "$ivl_for_loop0" 3 35, 3 35 0, S_0x138f04080;
 .timescale -9 -12;
v0x600001754090_0 .var/i "i", 31 0;
S_0x138f04f30 .scope begin, "$ivl_for_loop1" "$ivl_for_loop1" 3 36, 3 36 0, S_0x138f041f0;
 .timescale -9 -12;
v0x600001754000_0 .var/i "j", 31 0;
S_0x138f050a0 .scope module, "uut" "example" 3 10, 4 6 0, S_0x138f04080;
 .timescale -9 -12;
    .port_info 0 /INPUT 1 "clk_in";
    .port_info 1 /INPUT 1 "rst_in";
    .port_info 2 /INPUT 6 "a_in";
    .port_info 3 /INPUT 6 "b_in";
    .port_info 4 /OUTPUT 5 "c_out";
v0x600001754120_0 .net "a_in", 5 0, v0x6000017543f0_0;  1 drivers
v0x6000017541b0_0 .net "b_in", 5 0, v0x600001754480_0;  1 drivers
v0x600001754240_0 .var "c_out", 4 0;
v0x6000017542d0_0 .net "clk_in", 0 0, v0x6000017545a0_0;  1 drivers
v0x600001754360_0 .net "rst_in", 0 0, v0x600001754630_0;  1 drivers
E_0x600003051b00 .event posedge, v0x6000017542d0_0;
    .scope S_0x138f050a0;
T_0 ;
    %wait E_0x600003051b00;
    %load/vec4 v0x600001754360_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_0.0, 8;
    %pushi/vec4 0, 0, 5;
    %assign/vec4 v0x600001754240_0, 0;
    %jmp T_0.1;
T_0.0 ;
    %load/vec4 v0x6000017541b0_0;
    %load/vec4 v0x600001754120_0;
    %cmp/u;
    %flag_mov 8, 5;
    %jmp/0 T_0.2, 8;
    %load/vec4 v0x600001754240_0;
    %jmp/1 T_0.3, 8;
T_0.2 ; End of true expr.
    %load/vec4 v0x600001754240_0;
    %addi 1, 0, 5;
    %jmp/0 T_0.3, 8;
 ; End of false expr.
    %blend;
T_0.3;
    %assign/vec4 v0x600001754240_0, 0;
T_0.1 ;
    %jmp T_0;
    .thread T_0;
    .scope S_0x138f04080;
T_1 ;
    %delay 5000, 0;
    %load/vec4 v0x6000017545a0_0;
    %nor/r;
    %store/vec4 v0x6000017545a0_0, 0, 1;
    %jmp T_1;
    .thread T_1;
    .scope S_0x138f04080;
T_2 ;
    %vpi_call/w 3 23 "$dumpfile", "example.vcd" {0 0 0};
    %vpi_call/w 3 24 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x138f04080 {0 0 0};
    %vpi_call/w 3 25 "$display", "Starting Sim" {0 0 0};
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x6000017545a0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x600001754630_0, 0, 1;
    %pushi/vec4 0, 0, 6;
    %store/vec4 v0x6000017543f0_0, 0, 6;
    %pushi/vec4 0, 0, 6;
    %store/vec4 v0x600001754480_0, 0, 6;
    %delay 10000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x600001754630_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x600001754630_0, 0, 1;
    %vpi_call/w 3 34 "$display", "a_in    b_in    c_out" {0 0 0};
    %fork t_1, S_0x138f041f0;
    %jmp t_0;
    .scope S_0x138f041f0;
t_1 ;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x600001754090_0, 0, 32;
T_2.0 ;
    %load/vec4 v0x600001754090_0;
    %cmpi/s 256, 0, 32;
    %jmp/0xz T_2.1, 5;
    %fork t_3, S_0x138f04f30;
    %jmp t_2;
    .scope S_0x138f04f30;
t_3 ;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x600001754000_0, 0, 32;
T_2.2 ;
    %load/vec4 v0x600001754000_0;
    %cmpi/s 256, 0, 32;
    %jmp/0xz T_2.3, 5;
    %load/vec4 v0x600001754090_0;
    %pad/s 6;
    %store/vec4 v0x6000017543f0_0, 0, 6;
    %load/vec4 v0x600001754000_0;
    %pad/s 6;
    %store/vec4 v0x600001754480_0, 0, 6;
    %delay 10000, 0;
    %vpi_call/w 3 40 "$display", "%8b %8b %5c", v0x6000017543f0_0, v0x600001754480_0, v0x600001754510_0 {0 0 0};
    %load/vec4 v0x600001754000_0;
    %addi 1, 0, 32;
    %store/vec4 v0x600001754000_0, 0, 32;
    %jmp T_2.2;
T_2.3 ;
    %end;
    .scope S_0x138f041f0;
t_2 %join;
    %load/vec4 v0x600001754090_0;
    %addi 1, 0, 32;
    %store/vec4 v0x600001754090_0, 0, 32;
    %jmp T_2.0;
T_2.1 ;
    %end;
    .scope S_0x138f04080;
t_0 %join;
    %vpi_call/w 3 43 "$display", "Finishing Sim" {0 0 0};
    %vpi_call/w 3 44 "$finish" {0 0 0};
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "-";
    "example_tb.sv";
    "example.sv";
